tests/test_temperature.py

```diff
@@ -889,6 +889,32 @@ def test_nan_presence(self, tasmax_series):
         np.testing.assert_array_equal(hwi, [np.nan])
 
 
+class TestHeatWaveMagnitude:
+    def test_simple(self, tasmax_series):
+        tx = np.zeros(366)
+        tx[:5] = np.array([30, 30, 30, 30, 30])
+        tx = tasmax_series(tx + K2C, start="1/1/2000")
+        hwm = atmos.heat_wave_magnitude(tx, freq="YS")
+        np.testing.assert_array_equal(hwm, [25])
+
+    def test_convert_units(self, tasmax_series):
+        tx = np.zeros(366)
+        tx[:5] = np.array([30, 30, 30, 30, 30])
+        tx = tasmax_series(tx, start="1/1/2000")
+        tx.attrs["units"] = "C"
+        hwm = atmos.heat_wave_magnitude(tx, freq="YS")
+        np.testing.assert_array_equal(hwm, [25])
+
+    def test_nan_presence(self, tasmax_series):
+        tx = np.zeros(366)
+        tx[:5] = np.array([30, 30, 30, 30, 30])
+        tx[-1] = np.nan
+        tx = tasmax_series(tx + K2C, start="1/1/2000")
+
+        hwm = atmos.heat_wave_magnitude(tx, freq="YS")
+        np.testing.assert_array_equal(hwm, [np.nan])
```

Comment:
`heat_wave_max_length` would give a length of 5, not `np.nan`, I think it should also give a non-null result (25 in this case). However, I'm confused, I'm not sure what part of your code would produce `np.nan` instead of 25, do you know?

Reply:
I am not sure to be honest. I took this part from `TestHeatWaveIndex` and have assumed it follows some similar boilerplate checks that make the function return `np.nan`.

The following is the part that I then just adapted for magnitude.
```python

class TestHeatWaveIndex:

    ...

    def test_nan_presence(self, tasmax_series):
        tx = np.zeros(366)
        tx[:10] = np.array([29, 31, 31, 31, 29, 31, 31, 31, 31, 31])
        tx[-1] = np.nan
        tx = tasmax_series(tx + K2C, start="1/1/2000")

        hwi = atmos.heat_wave_index(tx, freq="YS")
        np.testing.assert_array_equal(hwi, [np.nan])
```

Reply:
Thanks, never mind. It must be the difference between the index, which is a simple function:
`xclim.indices.heat_wave_index`
and the class which performs a series of checks. 
`xclim.atmos.heat_wave_index`
I didn't realize it would not allow too many nans. 

Sorry about that!